### PR TITLE
Add Wear pairing flow and settings sync.

### PR DESCRIPTION
**Pull request by yschimke**
Use Horologist helpers for data syncing, including installing the wear app.  Then share the theme as proto.

To follow up:

- Some screens are not as readable, see video.

[weartheme.webm](https://user-images.githubusercontent.com/231923/226165913-244bfecd-8aea-4d55-a7d7-44a357861439.webm)

### CODE REVIEW

**Comment by yschimke** [comment]
@joreilly any idea about the build failure? How do I have common wire proto shared? If I run the proto step manually, then it works from then on.

I suspect Wire doesn't handle KMM perfectly yet.

**Comment by joreilly** [comment]
Hmm, not sure tbh.  @martinbonnin in case you've come across this?  Maybe some expect/actual setup needed to handle this.

**Comment by martinbonnin** [comment]
Ping @oldergod. Is KMM supported with Wire? Or do we need to wire (pun intended) the task manually?

**Comment by yschimke** [comment]
The tasks for wire look inconsistent. I might try moving to commonMain

**Comment by oldergod** [comment]
Currently, only protos in `commonMain` will be detected yes.

File: androidApp/src/main/java/dev/johnoreilly/confetti/di/AppModule.kt [hunk 1]
```diff
@@ -18,29 +18,24 @@ import dev.johnoreilly.confetti.speakerdetails.SpeakerDetailsViewModel
...
-        Authentication()
-    }
-    single<TokenProvider> {
-        object : TokenProvider {
-            override suspend fun token(forceRefresh: Boolean): String? {
-                return get<Authentication>().idToken(forceRefresh)
-            }
-        }
-    }
+    singleOf(::WorkManagerConferenceRefresh).withOptions { bind<ConferenceRefresh>() }
```
Comment (yschimke) [inline androidApp/src/main/java/dev/johnoreilly/confetti/di/AppModule.kt:36]: Ooops

**Comment by oldergod** [comment]
And I don't think Wire is gonna support other platforms for we'd have to generate expects/actuals. Maybe we could detect protos in `<platform>Main` and generate it all in `commonMain`, I don't know.

**Comment by yschimke** [comment]
@oldergod should I put this in a non KMP target?  Still not finding things in commonMain/proto

**Comment by oldergod** [comment]
There's a current bug which might be the cause https://github.com/square/wire/issues/2411
I'm in the middle of a PR which should improve everything https://github.com/square/wire/pull/2412 which I hope to land very soon™

**Comment by yschimke** [comment]
@oldergod Thanks, that was it.  Slightly different fix.

```
// https://github.com/square/wire/issues/2411
val wireTask = tasks.withType<com.squareup.wire.gradle.WireTask>()
tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
    dependsOn(wireTask)
}
afterEvaluate {
    wireTask.forEach {
        if ("generateCommon.+MainProtos".toRegex().matches(it.name)) {
            it.enabled = false
        }
    }
}
```

**Comment by yschimke** [comment]
cc @garanj

**Pull request merged**